### PR TITLE
Eric m/navbar

### PR DESCRIPTION
**Pull request by natheraf**
Used MUI to create a drop down for data pages. I did not want the Menu, navbar, to constantly reload on every page so I removed it on all pages and moved it to App so it was persistent. I moved the Menu because the drop down would reset and loose it's state when loading another page or the page refreshed.

### CODE REVIEW

File: client/src/components/Menu.tsx [hunk 1]
```diff
@@ -1,30 +1,47 @@
...
+import FormControl from '@mui/material/FormControl';
+import Select, { SelectChangeEvent } from '@mui/material/Select';
 
 const linkStyle = { color: '#343434', textDecoration: 'none' }
 const activeLinkStyle = { ...linkStyle, backgroundColor: '#eee', borderRadius: '15px', padding: '10px' }
 const links = [
     { title: 'Home', to: '/', activeSet: new Set(['/']) },
     { title: 'About', to: '/#about', activeSet: new Set(['/#about']) },
     { title: 'Services', to: '/#services', activeSet: new Set(['/#services']) },
+    { title: 'Contact', to: '/#contact', activeSet: new Set(['/#contact']) },
```
Comment (adbahrani) [inline client/src/components/Menu.tsx:14]: 👍

File: client/src/components/Menu.tsx [hunk 2]
```diff
@@ -1,30 +1,47 @@
...
 ]
 
 export const Menu = () => {
-    const location = useLocation()
-    const [activeLink, setActiveLink] = useState('')
+    const location = useLocation();
+    const [activeLink, setActiveLink] = useState('');
+    const navigate = useNavigate();
+    const [currentPage, setCurrentPage] = useState('');
+    const isLinkAnchor = (linkTo: string) => linkTo === '/' || linkTo.length >= 2 && linkTo.substring(0, 2) === '/#';
```
Comment (adbahrani) [inline client/src/components/Menu.tsx]: I believe you can simplify this logic a bit. 
You can utilize link.startWith("/#") to till if it's a section navigator, or even change the structure of links if needed

File: client/src/components/Menu.tsx [hunk 3]
```diff
@@ -57,13 +74,27 @@ export const Menu = () => {
...
                             <li key={link.title}>
-                                <Link to={link.to} style={activeLink === link.to ? activeLinkStyle : linkStyle}>
+                                <Link to={link.to} style={activeLink === link.to ? activeLinkStyle : linkStyle } onClick={clearDropdown}>
                                     {link.title}
                                 </Link>
                             </li>
                         ))}
+                        <FormControl sx={{minWidth: 150, maxHeight: 0, marginTop: -1}} size="small">
+                            <InputLabel id="dropdown-link-select- label" style={{color: 'black'}}>My Data</InputLabel>
+                            <Select
```
Comment (adbahrani) [inline client/src/components/Menu.tsx]: I'd change the default label to be more meaningful
e.g. Analytics Data

File: client/src/components/Menu.tsx [hunk 4]
```diff
@@ -57,13 +74,27 @@ export const Menu = () => {
...
+                        <FormControl sx={{minWidth: 150, maxHeight: 0, marginTop: -1}} size="small">
+                            <InputLabel id="dropdown-link-select- label" style={{color: 'black'}}>My Data</InputLabel>
+                            <Select 
+                                labelId="dropdown-link-select-label"
+                                id="dropdown-link-select"
+                                value={currentPage}
+                                label="MyData"
+                                onChange={handleChange}
+                            >
+                                {links.filter(link => !isLinkAnchor(link.to)).map(link => (
```
Comment (adbahrani) [inline client/src/components/Menu.tsx]: We probably want to change the styling for the drop down
Having an outline is a bit odd e.g.
you can remove that by using
 sx={{ boxShadow: 'none', '.MuiOutlinedInput-notchedOutline': { border: 0 } }}

**Review by adbahrani — COMMENTED** [review]
Nice work! Moving Menu to common place is pretty useful
Some adjustment to the dropdown and we should be set with that

File: client/src/components/Menu.tsx [hunk 5]
```diff
@@ -57,13 +74,27 @@ export const Menu = () => {
...
+                                    <MenuItem value={link.to}>{link.title}</MenuItem>
+                                ))}
+                            </Select>
+                        </FormControl>
```
Comment (adbahrani) [inline client/src/components/Menu.tsx:113]: Make sure you use prettier for formatting the file

**Comment by natheraf** [comment]
Would you like the `Data Analytics` label that goes above the user's choice be gone?

**Comment by adbahrani** [comment]
> Would you like the `Data Analytics` label that goes above the user's choice be gone?

Should  fine for now, thanks

File: client/src/components/Menu.tsx [hunk 6]
```diff
@@ -21,27 +21,27 @@ const links = [
...
-    const navigate = useNavigate();
-    const [currentPage, setCurrentPage] = useState('');
-    const isLinkAnchor = (linkTo: string) => linkTo === '/' || linkTo.startsWith('/#');
-    
+    const location = useLocation()
+    const [activeLink, setActiveLink] = useState('')
+    const navigate = useNavigate()
+    const [currentPage, setCurrentPage] = useState('')
+    const isLinkAnchor = (linkTo: string) => linkTo === '/' || linkTo.startsWith('/#')
+
```
Comment (adbahrani) [inline client/src/components/Menu.tsx:29]: Looks like there's a setting mismatch in your prettier, not a big issue for now, we can touch base on that next

**Review by adbahrani — APPROVED** [review]

**Pull request merged**